source/features/locked-issue.tsx

```diff
@@ -16,23 +16,25 @@ function LockedIndicator(): JSX.Element {
 	);
 }
 
-function addLock(element: HTMLElement): void {
-	element.after(
-		<LockedIndicator className="mb-2 rgh-locked-issue"/>,
-	);
-}
-
-function addStickyLock(element: HTMLElement): void {
-	element.after(
-		<LockedIndicator className="mr-2 mb-2 rgh-locked-issue"/>,
-	);
+function processTimeline(): void {
+	// Gather lock related events and sort by event ID
+	const events: HTMLElement[] = $$('.js-discussion .TimelineItem:has(.octicon-key, .octicon-lock)').sort((a, b) => b.id.localeCompare(a.id));
+	// If the most recent lock change event is locked
+	if (elementExists('.octicon-lock', events[0])) {
+		$('.gh-header-meta:not(:has(.rgh-locked-issue)) > :first-child')?.after(
+			<LockedIndicator className="mb-2 rgh-locked-issue"/>,
+		);
+		$('.gh-header-sticky:not(:has(.rgh-locked-issue)) .flex-row > :first-child')?.after(
+			<LockedIndicator className="mr-2 mb-2 rgh-locked-issue"/>,
+		);
+	} else { // Remove existing badge otherwise
+		$('.gh-header-meta > .rgh-locked-issue')?.remove();
+		$('.gh-header-sticky .flex-row > .rgh-locked-issue')?.remove();
+	}
 }
 
 function init(signal: AbortSignal): void {
-	// If reactions-menu exists, then .js-pick-reaction is the second child
-	// Logged out users never have the menu, so they should be excluded
-	observe('.logged-in:has(.js-pick-reaction:first-child) .gh-header-meta > :first-child', addLock, {signal});
-	observe('.logged-in:has(.js-pick-reaction:first-child) .gh-header-sticky .flex-row > :first-child', addStickyLock, {signal});
+	observe('.js-discussion .js-timeline-item', processTimeline, {signal});
```

Comment:
I'm not ok with processing every item on the page for this just to support old GHE installations and definitely not logged-out users (which RGH doesn't support, but in this case we targeted them specifically because it putting **wrong** information very visibly on the page)

Reply:
🥹 🫶